Source/Dafny/Resolver.cs

```diff
@@ -11932,6 +11949,42 @@ private class RBranchExpr : RBranch {
           "Internal Error: Unknown failure type in :- statement");
         return;
       }
+
+      Expression lhsExtract = null;
+      if (expectExtract) {
+        Method caller = codeContext as Method;
+        if (caller != null && caller.Outs.Count == 0 && s.ExpectToken == null) {
+          reporter.Error(MessageSource.Resolver, s.Rhs.tok, "Expected {0} to have a Success/Failure output value",
+            caller.Name);
+          return;
+        }
+
+        lhsExtract = s.Lhss[0];
+        if (lhsExtract is ExprDotName lex) {
+          Expression id = makeTemp("recv", s, codeContext, lex.Lhs);
+          lhsExtract = new ExprDotName(lex.tok, id, lex.SuffixName, lex.OptTypeArguments);
+        } else if (lhsExtract is SeqSelectExpr lseq) {
+          Expression id = makeTemp("recv", s, codeContext, lseq.Seq);
+          Expression id0 = null;
+          Expression id1 = null;
+          if (lseq.E0 != null) {
+            id0 = makeTemp("idx", s, codeContext, lseq.E0);
+          }
+          if (lseq.E1 != null) {
+            id1 = makeTemp("idx", s, codeContext, lseq.E1);
+          }
```

Comment:
A `SeqSelectExpr` l-value always has `.E1 == null`. (A Dafny expression like `a[lo..hi]` or `a[..hi]` is not an l-value.)

Reply:
OK